provider/datagen/src/transform/collator/mod.rs

```diff
@@ -312,3 +307,40 @@ collation_provider!(
     },
     toml_data
 );
+
+use std::fs::{self, File};
+use std::io::BufReader;
+use std::io::Read;
+/// TODO move into cache
+use std::path::{Path, PathBuf};
+
+/// Helper function to open a file and return failures as a crate error.
+fn open_reader(path: &Path) -> Result<BufReader<File>, DataError> {
```

Comment:
Nit: I'd prefer if you kept these functions in `crate::transform::reader` until no one needs them anymore. Moving them around like this just means they may need to be moved again and again.

Reply:
This was still TODO, added the TOML cache here as well.